### PR TITLE
Scripts/Icecrown: Small event of the down Nerub'ar Broodkeep

### PR DESCRIPTION
**Pull request by HoustonCore**
**Changes proposed:**
- Nerub'ar Broodkeep must fall down after passing triggers

**Target branch(es):** 3.3.5

- [√ ] 3.3.5

**Tests performed:**

Build, tested in-game.
- Need more testing and feedback
- I do not know if Nerub'ar Broodkeep should attack in a mile attack
- And what cooldown should spells have
- How do I add a thread when I drop it?

[Look video](https://youtu.be/z_x-yrMOPwM)

**Follow-up:**
Sorry for my English, I'm waiting for constructive criticism, and not here is not so there is not so, I ask for help in solving the standards

### CODE REVIEW

**Comment by Aokromes** [comment]
SAI is perfect valid for trash mobs.

Anyway, the result aint blizzlike enough, if you jump irc i can share you a video.

File: src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp [hunk 1]
```diff
@@ -1043,6 +1055,80 @@ struct npc_darkfallen_tactician : public DarkFallenAI
...
+    }
+
+    void Initialize()
+    {
+        CastNet = false;
+    }
+
+    void Reset() override
+    {
+        events.Reset();
```
Comment (jackpoz) [inline src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp:1072]: shouldn't Reset() set CastNet to false too (which is the whole point of having Initialize() ) ?

File: src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp [hunk 2]
```diff
@@ -1043,6 +1055,80 @@ struct npc_darkfallen_tactician : public DarkFallenAI
...
+        switch (events.ExecuteEvent())
+        {
+            case EVENT_CRYPT_SCARABS:
+                me->CastSpell(me->GetVictim(), SPELL_CRYPT_SCARABS, false);
+                events.Repeat(3s, 5s);
+                break;
+            case EVENT_DARK_MENDING:
+                if (Unit* target = DoSelectLowestHpFriendly(30.0f, 100000))
+                {
+                    if (me->GetInstanceScript() && (me->GetMap()->GetSpawnMode() & 0 || me->GetMap()->GetSpawnMode() & 1))
```
Comment (jackpoz) [inline src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp:1110]: I'm curious about what you are trying to accomplish with ```me->GetMap()->GetSpawnMode() & 0```

File: src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp [hunk 3]
```diff
@@ -1043,6 +1055,80 @@ struct npc_darkfallen_tactician : public DarkFallenAI
...
+                        me->CastSpell(target, SPELL_DARK_MENDING_N);
+                    else
+                        me->CastSpell(target, SPELL_DARK_MENDING_HC);
+                }
+                events.Repeat(5s, 9s);
+                break;
+            case EVENT_WEB_WRAP:
+                if (Unit* target = SelectTarget(SelectTargetMethod::Random, 0, 40.0f, true))
+                me->CastSpell(target, SPELL_WEB_WRAP, false);
```
Comment (jackpoz) [inline src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp:1119]: some indentation issues here

**Comment by HoustonCore** [comment]
> SAI is perfect valid for trash mobs.
> 
> Anyway, the result aint blizzlike enough, if you jump irc i can share you a video.
> 
> It will close #25193

https://www.youtube.com/watch?v=hrds9ANgcb8 it's blizzlike.

**Comment by TranceAccount** [comment]
Doesn't your script make all 4 of them fall down at the same time, while in the retail video you show that the first two fall down while the second two (the ones further away) are not yet down?

**Comment by jackpoz** [comment]
please comment your code so it's clear why things are being done in the way you do them 😃

**Comment by HoustonCore** [comment]
> please comment your code so it's clear why things are being done in the way you do them 😃

Yes, I'll explain, I'm just a novice dev)

**Comment by HoustonCore** [comment]
> Doesn't your script make the first four of them fall down at the same time, while in the retail video you show that the first two fall down while the second group of four broodkeepers (the ones further away) are not going down yet?
> 
> Edit: Sorry I can't count.

On my video, 2 packs fall, on blizzlike 1, all because of my trigger distance, but this can be fixed.

File: src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp (hunk 2 above)
Comment (HoustonCore) [reply src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp:1110]: > I'm curious about what you are trying to accomplish with `me->GetMap()->GetSpawnMode() & 0`

I wanted to give the difference of heal in heroics and normality.

File: src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp (hunk 3 above)
Comment (HoustonCore) [reply src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp:1119]: > some indentation issues here

Ops.

File: src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp (hunk 1 above)
Comment (HoustonCore) [reply src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp:1072]: > shouldn't Reset() set CastNet to false too (which is the whole point of having Initialize() ) ?

I don't understand much here, I don't speak English very well. I didn't declare Initialize (); in Reset () so?

File: src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.h [hunk 4]
```diff
@@ -147,6 +150,8 @@ enum ICCreaturesIds
     NPC_MURADIN_BRONZEBEARD_QUEST               = 38607,
     NPC_UTHER_THE_LIGHTBRINGER_QUEST            = 38608,
     NPC_LADY_SYLVANAS_WINDRUNNER_QUEST          = 38609,
+    NPC_NERUBAR_BROODKEEPER                     = 36725,
+    NPC_NERUBAR_BROODKEEPER_H                   = 38058,
```
Comment (offl) [inline src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.h:154]: Do not use heroic entry, heroic entry is never spawned in game

File: src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp (hunk 1 above)
Comment (jackpoz) [reply src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp:1072]: let me try again:
Shouldn't CastNet set to false again on Reset() ? or is there a reason for that not to happen ?

Comment (HoustonCore) [reply src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp:1072]: And exactly, if it goes to the standard position, CastNet will not work)

File: src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp (hunk 2 above)
Comment (jackpoz) [reply src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp:1110]: but ```me->GetMap()->GetSpawnMode() & 0``` is not really "valid" C++ code

**Comment by jackpoz** [comment]
I would prefer to keep SAI if SAI works (we have a lot of people moving from C++ to SAI, if we start merging moving from SAI to C++ we will spend our next 10 years just going back and forth)

**Comment by HoustonCore** [comment]
> I would prefer to keep SAI if SAI works (we have a lot of people moving from C++ to SAI, if we start merging moving from SAI to C++ we will spend our next 10 years just going back and forth)

SAI works terribly, mobs do not stumble from above and cast from there. There is not even a similar blizzlike

**Comment by HoustonCore** [comment]
My opinion static raids and dungeons are better to write in C++ because break the logic of SAI and all the content will go on .. If you want I will close the PR.

File: src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp [hunk 5]
```diff
@@ -1043,6 +1055,80 @@ struct npc_darkfallen_tactician : public DarkFallenAI
...
+        if (type == 1 && data == 1)
+        {
+            me->GetMotionMaster()->MoveFall();
+            me->SetHover(false);
+            me->SetByteValue(UNIT_FIELD_BYTES_1, UNIT_BYTES_1_OFFSET_ANIM_TIER, 0);
+            if (!CastNet)
+            {
+                CastNet = true;
+                me->RemoveAllAuras();
+                DoCastSelf(SPELL_CAST_NET_STARTROOM, true);
```
Comment (offl) [inline src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp:1090]: You can make it simple and move everything here to **OnlyOnce** areatrigger script and restore SAI because so far I see no reason to script that creature in core

File: src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp (hunk 2 above)
Comment (HoustonCore) [reply src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp]: Done

File: src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.h (hunk 4 above)
Comment (HoustonCore) [reply src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.h]: Done

File: src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp (hunk 5 above)
Comment (HoustonCore) [reply src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp:1090]: Okey

**Comment by offl** [comment]
> > I would prefer to keep SAI if SAI works (we have a lot of people moving from C++ to SAI, if we start merging moving from SAI to C++ we will spend our next 10 years just going back and forth)
> 
> SAI works terribly, mobs do not stumble from above and cast from there. There is not even a similar blizzlike

Probably because you need to remove these lines:
```text
entryorguid  source_type      id    link  event_type  event_phase_mask  event_chance  event_flags  event_param1  event_param2  event_param3  event_param4  event_param5  action_type  action_param1  action_param2  action_param3  action_param4  action_param5  action_param6  target_type  target_param1  target_param2  target_param3  target_param4  target_x  target_y  target_z  target_o  comment                                                                               
-----------  -----------  ------  ------  ----------  ----------------  ------------  -----------  ------------  ------------  ------------  ------------  ------------  -----------  -------------  -------------  -------------  -------------  -------------  -------------  -----------  -------------  -------------  -------------  -------------  --------  --------  --------  --------  --------------------------------------------------------------------------------------
      36725            0       1       0          10                 0           100           30             0            60          1000          1000             1           38              0              0              0              0              0              0            0              0              0              0              0         0         0         0         0  Nerub'ar Broodkeeper - Within 0-60 Range Out of Combat LoS - Set In Combat With Zone  
      36725            0       6       0           0                 0           100          543          4000          4000             0             0             0           91              2              3              0              0              0              0            1              0              0              0              0         0         0         0         0  Nerub'ar Broodkeeper  - In combat no repeat - Remove Flag Hover                       
```

**Comment by HoustonCore** [comment]
Okey i close PR. if you want use SAI)

**Comment by jackpoz** [comment]
I would suggest to check the open issues and fix one of those as possible contribution, or discuss how to fix a new issue if any (or you can open a new issue on github and discuss there about possible way to fix it)

**Comment by HoustonCore** [comment]
> I would suggest to check the open issues and fix one of those as possible contribution, or discuss how to fix a new issue if any (or you can open a new issue on github and discuss there about possible way to fix it)

Thank you I'll try)

**Comment by HoustonCore** [comment]
Guys, I decided to re-open it, because I spent a lot of time on it and I want to finish it. I'll transfer the code to the trigger, but I'll need some help. And please clean the comments that would not mislead others.

File: src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp [hunk 6]
```diff
@@ -1043,6 +1055,76 @@ struct npc_darkfallen_tactician : public DarkFallenAI
...
+        Initialize();
+        events.Reset();
+        events.ScheduleEvent(EVENT_CRYPT_SCARABS, 1s, 3s);  // Crypt Scarabs
+        events.ScheduleEvent(EVENT_DARK_MENDING, 10s, 15s); // Dark Mending
+        events.ScheduleEvent(EVENT_WEB_WRAP, 6s, 8s);       // Web Wrap
+    }
+
+    void SetData(uint32 type, uint32 data) override
+    {
+        if (type == 1 && data == 1)
```
Comment (Keader) [inline src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp:1081]: magic numbers...
change it for a enum

File: src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp [hunk 7]
```diff
@@ -1043,6 +1055,76 @@ struct npc_darkfallen_tactician : public DarkFallenAI
...
+                    me->CastSpell(target, SPELL_WEB_WRAP, false);
+                events.Repeat(5s, 9s);
+            break;
+        }
+
+        DoMeleeAttackIfReady();
+    }
+
+    private:
+        bool CastNet;
```
Comment (Keader) [inline src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp]: use our codestyle for private variables
like: _castNet

**Comment by offl** [comment]
@HoustonCore we can't script every single creature in instances in core just because our DB system is not perfect. Actually core combat AI system is too not perfect. Core scripts are used only if something cannot be done in DB or for bosses.

Actually almost everything you wrote here can be scripted in DB even today but it will be not stable enough(especially areatriggers, they can be activated only once(btw you still need OnlyOnce AT scripts). Also those ATs are used to make NPCs come to their new positions and not only for those spiders, so 2 ATs should be handled separately. If you want I can try to check sniffs later to know what really happens and how we can implement it.

File: sql/updates/world/3.3.5/2020_xx_xx_xx_world.sql [hunk 8]
```diff
@@ -0,0 +1,5 @@
+-- Nerub'ar broodkeep start event
+DELETE FROM `areatrigger_scripts` WHERE `entry` IN (5611,5612);
+INSERT INTO `areatrigger_scripts` (`entry`, `ScriptName`) VALUES
+(5611, 'at_icc_nerubar_broodkeeper'),
+(5612, 'at_icc_nerubar_broodkeeper');
```
Comment (Keader) [inline sql/updates/world/3.3.5/2020_xx_xx_xx_world.sql:5]: empty space missing

File: src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.h [hunk 9]
```diff
@@ -36,6 +36,9 @@ enum ICSharedSpells
     // Deathbound Ward
     SPELL_STONEFORM                     = 70733,
 
+    // Nerubar Broodkeeper
+    SPELL_FLIGHT                        = 43775,
```
Comment (Keader) [inline src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.h:40]: uneeded now?

File: src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp [hunk 10]
```diff
@@ -169,6 +176,11 @@ enum ICCTimedEventIds
 
     // Invisible Stalker (Float, Uninteractible, LargeAOI)
     EVENT_SOUL_MISSILE,
+
+    // Nerubar Broodkeeper
+    EVENT_CRYPT_SCARABS,
+    EVENT_DARK_MENDING,
+    EVENT_WEB_WRAP
```
Comment (Keader) [inline src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp:183]: uneeded now?

File: src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp [hunk 11]
```diff
@@ -128,9 +128,16 @@ enum ICCSpells
 
     // Darkfallen Tactician
     SPELL_SHADOWSTEP = 70431,
-    SPELL_BLOOD_SAP = 70432
+    SPELL_BLOOD_SAP = 70432,
+
+    // Nerubar broodkeeper
+    SPELL_CAST_NET_STARTROOM    = 69887,
+    SPELL_CRYPT_SCARABS         = 70965,
+    SPELL_WEB_WRAP              = 70980
```
Comment (Keader) [inline src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp:136]: uneeded now?

File: src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp [hunk 12]
```diff
@@ -128,9 +128,16 @@ enum ICCSpells
...
 };
 
+#define SPELL_DARK_MENDING RAID_MODE<uint32>(72322, 72324)
```
Comment (Keader) [inline src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp]: can be handled in spelldifficulty_dbc table

File: src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp [hunk 13]
```diff
@@ -128,7 +128,8 @@ enum ICCSpells
 
     // Darkfallen Tactician
     SPELL_SHADOWSTEP = 70431,
-    SPELL_BLOOD_SAP = 70432
+    SPELL_BLOOD_SAP = 70432,
```
Comment (Keader) [inline src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp:131]: , hehe

**Comment by HoustonCore** [comment]
Fixed everything, updated the PR description

**Comment by HoustonCore** [comment]
> @HoustonCore we can't script every single creature in instances in core just because our DB system is not perfect. Actually core combat AI system is too not perfect. Core scripts are used only if something cannot be done in DB or for bosses.
> 
> Actually almost everything you wrote here can be scripted in DB even today but it will be not stable enough(especially areatriggers, they can be activated only once(btw you still need OnlyOnce AT scripts). Also those ATs are used to make NPCs come to their new positions and not only for those spiders, so 2 ATs should be handled separately. If you want I can try to check sniffs later to know what really happens and how we can implement it.

That would be very good.

**Comment by jackpoz** [comment]
@Keader : thanks for volunteering on helping this PR getting polished, fixed and merged.

File: src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp [hunk 14]
```diff
@@ -1570,6 +1570,29 @@ class at_icc_start_blood_quickening : public AreaTriggerScript
...
+
+        bool OnTrigger(Player* player, AreaTriggerEntry const* /*areaTrigger*/)
+        {
+            if (player->IsGameMaster())
+                return false;
+
+            std::list<Creature*> creatureList;
+            player->GetCreatureListWithEntryInGrid(creatureList, NPC_NERUBAR_BROODKEEPER, 100.0f);
+            for (std::list<Creature*>::iterator itr = creatureList.begin(); itr != creatureList.end(); ++itr)
+                if (Creature* creature = *itr)
```
Comment (jackpoz) [inline src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp:1586]: this null check might be not needed

Comment (Keader) [reply src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp:1586]: naah, change everything to foreach
```
for (Creature const& creature : creatureList)
```

Comment (HoustonCore) [reply src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp:1586]: It's bad?

```
for (std::list<Creature*>::iterator itr = creatureList.begin(); itr != creatureList.end(); ++itr)
                (*itr)->GetMotionMaster()->MoveFall(),
                (*itr)->SetHover(false),
                (*itr)->SetByteValue(UNIT_FIELD_BYTES_1, UNIT_BYTES_1_OFFSET_ANIM_TIER, 0),
```

Comment (HoustonCore) [reply src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp:1586]: i change wait commit.

Comment (Keader) [reply src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp:1586]: it will not work, missing { } around all instructions.
just make it easy for you, use a foreach, you will have Creature* pointer already
e.g.

```
for (Creature* creature : creatureList)
{
    creature->GetMotionMaster()->MoveFall();
    creature->SetHover(false);
    creature->SetByteValue(UNIT_FIELD_BYTES_1, UNIT_BYTES_1_OFFSET_ANIM_TIER, 0);
}
```

File: src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp [hunk 15]
```diff
@@ -1570,6 +1570,26 @@ class at_icc_start_blood_quickening : public AreaTriggerScript
...
+        at_icc_nerubar_broodkeeper() : AreaTriggerScript("at_icc_nerubar_broodkeeper") { }
+
+        bool OnTrigger(Player* player, AreaTriggerEntry const* /*areaTrigger*/)
+        {
+            if (player->IsGameMaster())
+                return false;
+
+            std::list<Creature*> creatureList;
+            GetCreatureListWithEntryInGrid(creatureList, player, NPC_NERUBAR_BROODKEEPER, 135.0f);
+            for (Creature* creature : creatureList)
```
Comment (Keader) [inline src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp:1585]: https://github.com/TrinityCore/TrinityCore/pull/25410#discussion_r484084693

File: src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp [hunk 16]
```diff
@@ -1570,6 +1570,26 @@ class at_icc_start_blood_quickening : public AreaTriggerScript
...
+    public:
+        at_icc_nerubar_broodkeeper() : AreaTriggerScript("at_icc_nerubar_broodkeeper") { }
+
+        bool OnTrigger(Player* player, AreaTriggerEntry const* /*areaTrigger*/)
+        {
+            if (player->IsGameMaster())
+                return false;
+
+            std::list<Creature*> creatureList;
+            GetCreatureListWithEntryInGrid(creatureList, player, NPC_NERUBAR_BROODKEEPER, 135.0f);
```
Comment (Keader) [inline src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp:1584]: just wondering.
in your first video, it show this AreaTrigger activing 4 spiders.
But your blizzard video, show it activating 2 spiders, 135 yards is not too much?

Comment (HoustonCore) [reply src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp:1584]: I'm experimenting with yards.

File: src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp (hunk 15 above)
Comment (HoustonCore) [reply src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp:1588]: Done

File: src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp (hunk 16 above)
Comment (HoustonCore) [reply src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp:1587]: I have tried all possible options from 110-130 yards and there is no perfect option.

**Comment by Keader** [comment]
Ok, i tested this PR today and i found some issues.
I fixed them, i will push when github mirror stop trolling me (is cloning your fork with 300 kbp/s ._.)

**Comment by HoustonCore** [comment]
> Ok, i tested this PR today and i found some issues.
> I fixed them, will push when github mirror stop trolling me (is cloning your fork with 300 kbp/s ._.)
> 

Ahah great Internet)) Thank you for your help I will wait for your help)

**Comment by Keader** [comment]
fuck, my rebase fail lul

File: src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp [hunk 17]
```diff
@@ -36,36 +36,36 @@
 enum ICCTexts
 {
     // Highlord Tirion Fordring (at Light's Hammer)
-    SAY_TIRION_INTRO_1 = 0,
-    SAY_TIRION_INTRO_2 = 1,
-    SAY_TIRION_INTRO_3 = 2,
-    SAY_TIRION_INTRO_4 = 3,
-    SAY_TIRION_INTRO_H_5 = 4,
-    SAY_TIRION_INTRO_A_5 = 5,
+    SAY_TIRION_INTRO_1          = 0,
```
Comment (Keader) [inline src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp:39]: @ccrs please, NEVER mess with enum identation again 😠

**Comment by HoustonCore** [comment]
Wow this is not a small fix)) looks great))

File: src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp [hunk 18]
```diff
@@ -1547,6 +1667,24 @@ class at_icc_start_blood_quickening : public AreaTriggerScript
...
+{
+    public:
+        at_icc_nerubar_broodkeeper() : OnlyOnceAreaTriggerScript("at_icc_nerubar_broodkeeper") { }
+
+        bool TryHandleOnce(Player* player, AreaTriggerEntry const* areaTrigger) override
+        {
+            if (InstanceScript* instance = player->GetInstanceScript())
+            {
+                if (player->IsGameMaster())
+                return false;
```
Comment (Carbenium) [inline src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp:1680]:
```suggestion
                    return false;
```

**Comment by Keader** [comment]
merging it in 24h

**Comment by HoustonCore** [comment]
> merging it in 24h

It looks very cool, but. If you move at normal speed, the first spiders [fall](https://www.youtube.com/watch?v=PePgvO1LkNQ) 
And if you move at a fast speed .mod speed 20, then the event works [perfectly](https://youtu.be/Dob7Hu_iKt8).

**Comment by jackpoz** [comment]
>// TODO: Remove it when Anim PR get merge, Disable/EnableGravity will handle with it

how about the anim PR gets merged first, this PR gets updated, tested and merged after ?

**Comment by Keader** [comment]
> > merging it in 24h
> 
> It looks very cool, but. If you move at normal speed, the first spiders [fall](https://www.youtube.com/watch?v=PePgvO1LkNQ)
> And if you move at a fast speed .mod speed 20, then the event works [perfectly](https://youtu.be/Dob7Hu_iKt8).

video is private, i cant see



> > // TODO: Remove it when Anim PR get merge, Disable/EnableGravity will handle with it
> 
> how about the anim PR gets merged first, this PR gets updated, tested and merged after ?

PR will just handle with anim automatically, instead force it using 
```cpp
me->SetByteValue(UNIT_FIELD_BYTES_1, UNIT_BYTES_1_OFFSET_ANIM_TIER, Animation);
```
So... no pratical change at all, we just need remove this SetByteValue when pr gets merged (i will do it)

**Comment by HoustonCore** [comment]
@Keader ops I open video.

**Comment by Keader** [comment]
@HoustonCore try now

**Comment by HoustonCore** [comment]
> @HoustonCore try now

I'll check later, I'm at work.

**Comment by Keader** [comment]
Ty for your pr 👍

**Comment by HoustonCore** [comment]
> Ty for your pr 👍 

Thank you. You did the biggest job)
